Feat/delete releated media

cloedu87 opened the pull request:
referential deletion of s3 stored media throughout the collection lifecycle

Solves
- [x] [SCN-386](https://infofauna-support.atlassian.net/browse/SCN-386)
- [x] [SCN-455](https://infofauna-support.atlassian.net/browse/SCN-455)
- [x] [SCN-475](https://infofauna-support.atlassian.net/browse/SCN-475)

at resource destruction: ensures referential integrity not only between database related objects, but as well takes care to delete all images and files stored on s3 which would not be automatically deleted

cloedu87 committed:
feat: introduce usage rules; delete referenced medias

cloedu87 committed:
feat: remove unnecessary delete actions; add delete related medias
change; generated rules.md for AI hype heads :-)

cloedu87 committed:
feat: new modal confirmation text; test fixes for new deletion
constraints; update docs; update gettext

cloedu87 committed:
feat: new delete record and collection confirmation texts

cloedu87 committed:
chore: merge in develop

cloedu87 committed:
Merge branch 'develop' into feat/delete-releated-media

cloedu87 committed:
fix: mail validation

cloedu87 committed:
fix: published_records > migrations to have constraints conditinally
deleted, before
recreating them;

cloedu87 committed:
feat: referential deletion of s3 stored media throughout the collection
lifecycle

sidler92 reviewed (dismissed):
looks good :birthday:

cloedu87 committed:
feat: make images deletable; make referential integrity work from
collection over image_upload and images down to attachments and the
files on s3

sidler92 reviewed (dismissed):
very nice carousel :carousel_horse: looks good

cloedu87 commented:
> very nice carousel 🎠 looks good

thx, sure... just don't show to @luchszebbra ;-)

cloedu87 committed:
chore: dependency updates

cloedu87 committed:
chore: adjust regex to be compatible with otp 28

cloedu87 committed:
chore: update docker alpine version

cloedu87 committed:
chore: Merge remote-tracking branch 'origin/develop' into feat/delete-releated-media

sidler92 reviewed (dismissed):
:tada:

cloedu87 committed:
chore: elixir 18 on alpine 3.22 is not yet available. reverting
